routing_common/routing_common_tests/transit_test.cpp

```diff
@@ -17,23 +17,70 @@ namespace routing
 {
 namespace transit
 {
-template<class Obj>
-void TestSerialization(Obj const &obj)
+template<class S, class D, class Obj>
+void TestCommonSerialization(Obj const & obj)
 {
   vector<uint8_t> buffer;
   MemWriter<vector<uint8_t>> writer(buffer);
 
-  Serializer<MemWriter<vector<uint8_t>>> serializer(writer);
+  S serializer(writer);
   obj.Visit(serializer);
 
   MemReader reader(buffer.data(), buffer.size());
   ReaderSource<MemReader> src(reader);
   Obj deserializedObj;
-  Deserializer<ReaderSource<MemReader>> deserializer(src);
+  D deserializer(src);
   deserializedObj.Visit(deserializer);
 
   TEST(obj.IsEqualForTesting(deserializedObj), (obj, deserializedObj));
 }
+
+void TestSerialization(TransitHeader const & header)
+{
+  TestCommonSerialization<FixedSizeSerializer<MemWriter<vector<uint8_t>>>,
+                          FixedSizeDeserializer<ReaderSource<MemReader>>>(header);
+}
+
+template<class Obj>
+void TestSerialization(Obj const & obj)
+{
+  TestCommonSerialization<Serializer<MemWriter<vector<uint8_t>>>,
+                          Deserializer<ReaderSource<MemReader>>>(obj);
+}
+
+UNIT_TEST(Transit_HeaderRewriting)
+{
+  TransitHeader const bigHeader(1 /* version */, 1000 /* gatesOffset */, 200000 /* edgesOffset */,
+                                300000 /* transfersOffset */, 400000 /* linesOffset */,
+                                5000000 /* shapesOffset */, 6000000 /* networksOffset */,
+                                700000000 /* endOffset */);
+
+  TransitHeader header;
+  vector<uint8_t> buffer;
+  MemWriter<vector<uint8_t>> writer(buffer);
+
+  // Writing.
+  uint64_t constexpr startOffset = 0;
```

Comment:
By "constant propagation" I meant to get rid of this variable.

Reply:
Done